File: solutions/scala/etl/2/src/main/scala/Etl.scala
object Etl {
  def transform(scoreMap: Map[Int, Seq[String]]): Map[String, Int] =
    // scoreMap
    // .flatMap({ case (score, chars) => chars.map(_.toLowerCase -> score) })
    for {
      (score, chars) <- scoreMap
      letter <- chars
    } yield letter.toLowerCase() -> score
}

File: solutions/elixir/etl/1/lib/etl.ex
defmodule ETL do
  @doc """
  Transforms an old Scrabble score system to a new one.

  ## Examples

    iex> ETL.transform(%{1 => ["A", "E"], 2 => ["D", "G"]})
    %{"a" => 1, "d" => 2, "e" => 1, "g" => 2}
  """
  @spec transform(map) :: map
  def transform(input) do
    input
    |> Enum.map(fn {points, chars} ->
      Map.new(chars, fn ch -> {String.downcase(ch), points} end)
    end)
    |> Enum.reduce(&Map.merge/2)
  end
end
